src/codegen/compile.ml

```diff
@@ -5687,7 +5687,8 @@ module MakeSerialization (Strm : Stream) = struct
           Heap.store_field MutBox.field
         )
       | Non ->
-        E.trap_with env "IDL error: deserializing value of type None"
+        skip get_idltyp ^^
+        coercion_failed "IDL error: deserializing value of type None"
```

Comment:
This looks wrong; Non is a different type than future types. Maybe the type table parser returns Non for future types and should return Future

Reply:
So the reason I added this is that your new future type tests
```
// Future types
// This uses 0x67 for the “future type”; bump (well, decrement) once that
// becomes a concrete future type
assert blob "DIDL\01\67\00\02\00\7e\00\00\01" == "(null, true)" : (opt empty,bool) "skipping minimal future type";
assert blob "DIDL\01\67\03ABC\02\00\7e\05\00hello\01" == "(null,true)" : (opt empty,bool) "skipping future type with data";
```
need to be able to deserialise at `opt empty` (? None) and fail when deserialization of the None value fails. Are you sure that's wrong then - seems right to me.

Reply:
Ah, yes, you are right: This not actually about our handling of future types, but just a bug in our subtyping of `Non`.